src/Controller.cpp

```diff
@@ -84,7 +84,30 @@ void Controller::setFrameHandler(void (*fptr)(CANFrame *msg), byte opcodes[], by
 //
 void Controller::setParams(unsigned char *mparams)
 {
+  bool producer = 0;
+  bool consumer = 0;
+  bool nPreNormal = 0;
+  bool conOwn = 0;
```

Comment:
You never set conOwn here.
We talked about a COEService at some stage. Would this be a good idea? It may be responsible for putting produced events into the events table.

Reply:
I put it there as a place holder because it is a flag in params[8].  However, we know we'll need to come back to it when we decide how COE will be done so I'll leave it out, for now.